### PR TITLE
Change IMentoExchange and IReserve interfaces.

### PR DESCRIPTION
**Pull request by pedro-clabs**
To the agreed upon interfaces, so that #9780 can be merged.

### CODE REVIEW

**Comment by pedro-clabs** [comment]
Changing just the interface means that the implementations are now broken -- so please disregard test results.

Do we agree that that'll be the interface moving forward? If so, I'd suggest pushing changes to the implementation here and merging.

File: packages/protocol/contracts/stability/interfaces/IMentoExchange.sol [hunk 1]
```diff
@@ -6,26 +6,19 @@ pragma solidity ^0.5.13;
...
-   * @param tokenOut Address of the token being exchanged for.
+   * @notice Returns the output amount for a given input amount.
    * @param tokenInBucketSize Size of the tokenIn bucket.
    * @param tokenOutBucketSize Size of the tokenOut bucket.
    * @param amountIn Amount of tokenIn being paid in.
    * @return amountOut Amount of tokenOut that will be paid out.
-   * @return newtokenInBucketSize Size of the tokenIn bucket after the swap.
-   * @return newtokenOutBucketSize Size of the tokenOut bucket after the swap.
    */
   function getAmountOut(
```
Comment (bowd) [inline packages/protocol/contracts/stability/interfaces/IMentoExchange.sol:15]: We should also have `getAmountIn`

Comment (bowd) [reply packages/protocol/contracts/stability/interfaces/IMentoExchange.sol:15]: And we should remove `getUpdateBuckets`

**Review by bowd — CHANGES_REQUESTED** [review]

File: packages/protocol/contracts/stability/interfaces/IMentoExchange.sol (hunk 1 above)
Comment (bowd) [reply packages/protocol/contracts/stability/interfaces/IMentoExchange.sol:15]: We also don't need to pass in the token addresses.

Comment (bowd) [reply packages/protocol/contracts/stability/interfaces/IMentoExchange.sol:15]: And if you want to make this pass you can rename the existing interface into `IMentoExchangeLegacy` and use that in all existing implementations. And provide the new `IMentoExchange` as we need it. That way we can migrate to using the new interface in the respective implementations without turning the branch red.

**Pull request closed without merging**